python/mlc_llm/interface/gen_config.py

```diff
@@ -89,6 +90,17 @@ def gen_config(  # pylint: disable=too-many-locals,too-many-arguments,too-many-b
 ):
     """Entrypoint of MLC Chat configuration generation."""
     # Step 1. Initialize `mlc-chat-config.json` using `config.json`
+    conversation_reg = ConvTemplateRegistry.get_conv_template(conv_template)
+    if conversation_reg is None:
+        logger.warning(
+            "%s: Conversation template is not registered in ConvTemplateRegistry: %s",
+            red("Warning"),
+            conv_template,
+        )
+        conversation = conv_template  # type: ignore
+    else:
+        conversation = conversation_reg.to_json_dict()  # type: ignore
```

Comment:
Also, should we keep both fields for now to keep the `mlc-chat-config.json` backward compatible? This is mainly for WebLLM: imagine a user who has shipped a product using, say npm 0.2.27. If we update all `mlc-chat-config.json` on HF and remove the original `conv_template` field, then the user is forced to update WebLLM, otherwise it would break.

Reply:
What you mentioned is forward compatibility, and I think it is fine to not do that